sky/skylet/job_lib.py

```diff
@@ -341,21 +341,15 @@ def update_job_status(job_owner: str,
 
     job_client = _create_ray_job_submission_client()
 
-    def get_job_status(job_id) -> Optional[JobStatus]:
-        try:
-            # The return value is a string, e.g. 'RUNNING', which conflicts
-            # with the return type in ray code.
-            ray_status = job_client.get_job_status(job_id)
-            return _RAY_TO_JOB_STATUS_MAP[ray_status]
-        except RuntimeError as e:
-            # If the job does not exist or if the request to the
-            # job server fails.
-            if 'does not exist' in str(e):
-                return None
-            raise
+    ray_job_infos = job_client.list_jobs()
+    ray_statuses: List[JobStatus] = []
+    for ray_job_id in ray_job_ids:
+        if ray_job_id not in ray_job_infos:
```

Comment:
Interestingly, in Ray master branch `ray_job_infos` becomes a List[JobDetails]

https://github.com/ray-project/ray/blob/master/dashboard/modules/job/sdk.py#L225

Perhaps note in a comment it's a Dict of job id -> JobDetails as of 1.13.0.

Reply:
Good point! Fixed. Thanks!